src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -1486,6 +1486,11 @@ OBJECTREF* GcInfoDecoder::GetRegisterSlot(
     _ASSERTE(regNum >= 0 && regNum <= 14);
     _ASSERTE(regNum != 13);  // sp
 
+#ifdef FEATURE_REDHAWK
+    PTR_UIntNative* ppReg = &pRD->pR0;
+
```

Comment:
This needs to special case `regNum == 14`

Reply:
I have a question. I do not fully understand your remark. But after looking at the source code at least I think I can ask reasonable question.

1. What is `regNum == 14`? If I take `REGDISPLAY` from `src\coreclr\nativeaot\Runtime\regdisplay.h` this is `SP`.
2. In the assert above written that `_ASSERTE(regNum != 13);  // sp`, but 13 index is `pLR`.
3. Why special case? I take a look at `TARGET_ARM64` and they just look at plan structure. I probably miss some concept here.

Reply:
Looks like I got it. I found this definition and that's from where `regNum` values should be used.
https://github.com/dotnet/runtimelab/blob/dc24a19f55ebe8546bce15db4d208ce2b3053355/src/coreclr/nativeaot/libunwind/include/libunwind.h#L603-L605

Reply:
Right, the REGDISPLAY structure does not have SP in the usual position, and so the code has to compensate for everything above SP. It does not have much to with libunwind.h.